dockstore-webservice/src/main/java/io/dockstore/webservice/core/metrics/CostStatisticMetric.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.core.metrics;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "cost_metric")
+@Schema(name = "CostMetric", description = "This describes aggregated cost metrics for workflow executions in USD.")
+public class CostStatisticMetric extends StatisticMetric {
+    public static final String UNIT = "USD";
```

Comment:
Rather than use a string for the unit and a double for the value, I wonder if there is value in the JDK adjacent Java Money, just in case we allow for different units later
https://www.baeldung.com/java-money-and-currency
https://github.com/JavaMoney/jsr354-api

Some backstory at https://stackoverflow.com/questions/52998955/javas-monetary-and-currency-operation-jsr-354

Reply:
If we do allow different currencies later, I think we would still want to aggregate to one currency (example USD) for display purposes. If that's the use case, it seems like Java Money is something that the metrics aggregator would use instead of the webservice so that it can do conversions. For example, it would use the string unit and double value that the webservice recorded to create the `Money` object and do further processing

Reply:
Unsure, we may display different units to different users depending on their settings. 
For now though, I more meant it may be nice to validate the currency, like with ` CurrencyUnit term = TestCurrency.of("USD");` 
https://github.com/JavaMoney/jsr354-api/blob/1e0b77795a183b1f4dba7212df790a9df8a67977/src/test/java/javax/money/convert/ExchangeRate_BuilderTest.java#L63 in the constructor, so in the case that we add ingestion of different currencies to the endpoint, we will do some sanity checking on the unit.

Reply:
Actually, even using double for currency may be an issue.  
https://dzone.com/articles/never-use-float-and-double-for-monetary-calculatio

Reply:
For statistical purposes, it's probably ok to use `double`, as long as we're aware that it can't represent most decimal fractions exactly (an IEEE 754 mantissa is interpreted as base 2 fraction and thus can't exactly represent reduced fractions with non-power-of-2 denominators).  The main gotchas are comparisons and formatting for storage/output (beware the "1.100000000000045").  OTOH, if we were a bank, we'd probably prefer an exact representation.